tests/trainer/test_fsdp_checkpoint.py

```diff
@@ -890,3 +892,143 @@ def test_cleanup_sharded_checkpoints(
+@pytest.mark.filterwarnings(r'ignore:TypedStorage is deprecated.:UserWarning')
+@pytest.mark.filterwarnings(r'ignore:.*metrics are not saved with sharded state dict.*:UserWarning')
+def test_fsdp_planner(
+    world_size,
+    tmp_path: pathlib.Path,
+    weights_only: bool,
+    planner: Optional[str],
+    s3_bucket,
+    s3_ephemeral_prefix,
+    request,
+):
+
+    from torch.distributed.checkpoint._nested_dict import flatten_state_dict
+    from torch.distributed.checkpoint._sharded_tensor_utils import _flatten_sharded_tensors
+    from torch.distributed.checkpoint.default_planner import DefaultLoadPlanner, DefaultSavePlanner
+    from torch.distributed.checkpoint.metadata import STATE_DICT_TYPE, Metadata
+
+    class RenameSavePlanner(DefaultSavePlanner):
+
+        def set_up_planner(
+            self,
+            state_dict: STATE_DICT_TYPE,
+            is_coordinator: bool,
+        ) -> None:
+            # suffix all keys with `foo_``
+            state_dict['state']['model'] = {k + '_foo': v for k, v in state_dict['state']['model'].items()}
+
+            super().set_up_planner(state_dict, is_coordinator)
+
+    class RenameLoadPlanner(DefaultLoadPlanner):
+
+        def set_up_planner(
+            self,
+            state_dict: STATE_DICT_TYPE,
+            metadata: Metadata,
+            is_coordinator: bool,
+        ) -> None:
+            if 'state' not in state_dict:
+                super().set_up_planner(state_dict, metadata, is_coordinator)
+                return
+
+            self.original_state_dict = state_dict
+
+            state_dict = dict(state_dict.items())
+            state_dict['state'] = dict(state_dict['state'].items())
+            state_dict['state']['model'] = {k + '_foo': v for k, v in state_dict['state']['model'].items()}
+
+            if self.flatten_sharded_tensors:
+                state_dict = _flatten_sharded_tensors(state_dict)
+
+            if self.flatten_state_dict:
+                state_dict, self.mappings = flatten_state_dict(state_dict)
+
+            self.state_dict = state_dict
+            self.metadata = metadata
+            self.is_coordinator = is_coordinator
+
+    save_planner = planner
+    load_planner = planner
+    if planner == 'rename':
+        save_planner = RenameSavePlanner()
+        load_planner = RenameLoadPlanner()
+
+    run_name = None
+
+    tmp_paths = dist.all_gather_object(os.path.abspath(tmp_path))
+    save_folder = os.path.join(tmp_paths[0], 'checkpoints', '{run_name}')
+
+    save_filename = 'ba{batch}-rank{rank}.pt'
+
+    fsdp_config = FSDPConfig(
+        state_dict_type='sharded',
+        load_planner=load_planner,
+        save_planner=save_planner,
+    )
+
+    trainer1 = get_trainer(
+        save_folder=str(save_folder),
+        save_filename=save_filename,
+        run_name=run_name,
+        max_duration='2ba',
+        save_interval='2ba',
+        save_weights_only=weights_only,
+        fsdp_config=fsdp_config,
+    )
```

Comment:
```suggestion
    trainer1 = get_trainer(
        save_folder=str(save_folder),
        max_duration='2ba',
        save_interval='2ba',
        save_weights_only=weights_only,
        fsdp_config=fsdp_config,
    )
```
Removes args which are same value as defaults